packages/xdl/src/DevSession.js

```diff
@@ -33,25 +34,53 @@ export async function startSession(
     }
 
     try {
-      let url = await UrlUtils.constructManifestUrlAsync(projectRoot);
+      let platformSessions = [];
+      if (exp['platforms'].includes('web')) {
```

Comment:
Thanks for the great advice!

Reply:
`Project` already controls which servers to start up so it should pass that information along to `DevSession` (via calling one method instead of another, or by an argument) instead of having DevSession infer what Project's behavior was. This greatly reduces the spatial coupling between the two modules so that if we change Project's policy for starting servers, the logic in DevSession doesn't have to change much.